core/src/main/java/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -910,6 +917,7 @@ protected QuestionDef parseUpload(IFormElement parent, Element e, int controlUpl
         Vector usedAtts = new Vector();
         usedAtts.addElement("mediatype");
         usedAtts.addElement(REF_ATTR);
+        usedAtts.addElement("imageDimensionScaledMax");

```

Comment:
The only way that would really work would be to have parseControl return the used attributes for whatever extension parser was parsed within it, since there's no interaction with the extensions in this method. Are we ok with the awkwardness of parseControl() having to return like Pair<QuestionDef, List<UsedAtts>>?